### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by AlexPelevin99**

### CODE REVIEW

**Review by pavlejviki — CHANGES_REQUESTED** [review]
All checks should have been passed.

**Comment by AlexPelevin99** [comment]
one checks not passed because test has error
image

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,29 @@
...
+        self.age = age
+        self.people[self.name] = self
+
+    @classmethod
+    def find_by_name(cls, name: str) -> Person | None:
+        return cls.people[name]
+
+
+def create_person_list(people: list) -> list[Person]:
+    for p in people:
```
Comment (Dimosphen1) [inline app/main.py:18]: Such variable name as `p` is not allowed as stated by `flake8`. Replace it with a more descriptive one, e.g. `person`.

File: app/main.py [hunk 2]
```diff
@@ -1,8 +1,29 @@
...
+        Person(p["name"], p["age"])
 
+    for p in people:
```
Comment (Dimosphen1) [inline app/main.py:21]: Do not use single-letter variable names here.

File: app/main.py [hunk 3]
```diff
@@ -1,8 +1,29 @@
...
+    people: dict = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[self.name] = self
+
+    @classmethod
+    def find_by_name(cls, name: str) -> Person | None:
+        return cls.people[name]
```
Comment (Dimosphen1) [inline app/main.py:14]: It would be better to use `.get()` here thus making code less error-prone.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Resolve issues raised by `flake8`.

File: app/main.py [hunk 4]
```diff
@@ -1,8 +1,29 @@
+from __future__ import annotations
```
Comment (Dimosphen1) [inline app/main.py:1]: Do not use this import here, it breaks the tests.

File: app/main.py [hunk 5]
```diff
@@ -1,8 +1,29 @@
...
+    people: dict = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[self.name] = self
+
+    @classmethod
+    def find_by_name(cls, name: str) -> Person | None:
+        return cls.people.get(name)
```
Comment (Dimosphen1) [inline app/main.py:14]: There is no need for this classmethod, you should get value from `people` attribute from the `Person` class directly.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested, check tests before adding changes.

**Pull request closed without merging**